Themis/Models/Repository.swift

```diff
@@ -132,6 +135,53 @@ class Node: Hashable, ObservableObject {
     func hash(into hasher: inout Hasher) {
         hasher.combine(path)
     }
+
+    @MainActor
+    public func calculateDiff(templateParticipationId: Int) async {
+        // only calculate once
```

Comment:
I feel like these checks should be in a separate function so that this one is shorter and you still can re-evaluate manually if needed 🤔

Reply:
I mean half of the validation also consists of unwrapping, which definitely makes more sense to do in the same function. And if the situation should really appear (which I don't think it will), we can use the now introduced diffCalculated variable.

Reply:
Ok, fine 👍